teps/0147-tekton-artifacts-phase1.md

```diff
@@ -0,0 +1,285 @@
+---
+status: implementable
+title: Tekton Artifacts phase 1
+creation-date: '2024-01-24'
+last-updated: '2024-01-24'
+authors:
+- '@chitrangpatel'
+contributors:
+- '@afrittoli'
+- '@wlynch'
+---
+
+# TEP-0147: Tekton Artifacts phase 1
+---
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+    - [Phase 1](#phase-1)
+    - [Phase 2](#phase-2)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+  - [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Artifact Provenance Data](#artifact-provenance-data)
+  - [From Step into Status](#from-step-into-status)
+  - [From Status into Provenance](#from-status-into-provenance)
+- [Design Details](#design-details)
+- [Design Evaluation](#design-evaluation)
+  - [Reusability](#reusability)
+  - [Simplicity](#simplicity)
+  - [Flexibility](#flexibility)
+  - [Conformance](#conformance)
+  - [User Experience](#user-experience)
+  - [Performance](#performance)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Implementation Plan (TBD)](#implementation-plan-tbd)
+  - [Test Plan](#test-plan)
+  - [Infrastructure Needed](#infrastructure-needed)
+  - [Upgrade and Migration Strategy](#upgrade-and-migration-strategy)
+  - [Implementation Pull Requests](#implementation-pull-requests)
+  - [References](#references)
+<!-- /toc -->
+
+## Summary
+
+[TEP-0139](https://github.com/tektoncd/community/blob/main/teps/0139-trusted-artifacts.md) introduces the Artifacts API for the use case of artifacts (backed to a Persistent Volume) that are produced and consumed within the provenance. After more reflection and discussions, we think that it is only part of the use case of the artifacts API. It does not cover the use case of artifacts like `images`, `packages`, `source code` as discussed in [this feature request](https://github.com/tektoncd/pipeline/issues/6326). We want to be able to handle different artifact types using the same API. With `StepActions` now in play, that design needs to be updated to include `StepActions`. While the Artifact API design is still undergoing early design and brainstorming, one aspect that has general agreement is around reporting of the artifact provenance data to the `TaskRun` `Status`. We therefore think that it will be nice to split the design of Artifacts support into multiple phases and incrementally add features as needed. For now, we have identified two phases to completely capture the requirements and use cases captured in [TEP-0139](https://github.com/tektoncd/community/blob/main/teps/0139-trusted-artifacts.md) and [this feature request](https://github.com/tektoncd/pipeline/issues/6326). 
```

Comment:
Just a comment (no action required).
External inputs and outputs were set as a [non-goal](https://github.com/tektoncd/community/blob/main/teps/0139-trusted-artifacts.md#non-goals) in TEP-0139 to limit the scope of the TEP. The intention was to eventually extend that to cover such artifacts as well. The provenance format defined in TEP-0139 selected `purl` as a format with that vision in mind.

Compared to TEP-0139, this one expands the provenance format to other formats, and excludes almost everything else that was there, to move it to phase-2 or non-goal.